frontend/components/portfolio/initiative/membership/MenuButton.vue

```diff
@@ -0,0 +1,162 @@
+      hoverText = tt('Remove all portfolios from initiative')
+    } else if (anySelected) {
+      icon = 'partial'
+      addIfClick = true
+      hoverText = tt('Add unselected portfolios to initiative')
+    }
+    return {
+      id: initiative.id,
+      label: initiative.name,
+      icon,
+      cmd: changeMemberships(initiative.id, addIfClick),
+      hoverText,
+      created: initiative.createdAt,
+    }
+  })
+  // Created is an ISO date time string. This sorts by newest first, without having
+  // to parse the date.
+  result.sort((a, b) => a.created < b.created ? 1 : -1)
+  return result
+})
+</script>
+
+<template>
+  <div>
+    <PVButton
+      :disabled="!props.selectedPortfolios || props.selectedPortfolios.length === 0"
+      class="p-button-sm"
+      :class="visible ? '' : 'p-button-outlined'"
+      :label="tt('Initiative Memberships') + selectedCountSuffix(props.selectedPortfolios)"
+      icon="pi pi-sitemap"
+      @click="toggleMenu"
+    />
+    <PVOverlayPanel
+      ref="overlayPanel"
+      :pt="{ content: { class: 'p-0' } }"
+      @hide="() => { visible = false }"
+      @show="() => { visible = true }"
+    >
+      <div class="flex flex-column align-items-stretch">
+        <div class="font-bold text-xl p-3 border-bottom-1 border-600 flex gap-2 align-items-center">
+          <span>{{ tt('Initiative Memberships') }}</span>
+          <PVButton
+            icon="pi pi-times"
+            class="p-button-text px-1 py-0 w-auto h-auto p-button-secondary"
+            @click="toggleMenu"
+          />
+        </div>
+        <div
+          v-for="option in initiativeOptions"
+          :key="option.id"
+          class="border-bottom-1 border-400"
+        >
+          <PVButton
+            v-tooltip="option.hoverText"
+            class="text-left p-button-text w-full"
+            @click="option.cmd"
+          >
+            <div class="flex justify-content-start align-items-center gap-3">
+              <div
+                class="pseudo-checkbox flex-0 border-2 border-round flex justify-content-center align-items-center"
+                :class="option.icon === 'empty' ? 'bg-white' : 'bg-primary-500 text-white border-primary-500'"
+              >
+                <i
+                  v-if="option.icon === 'full'"
+                  class="pi pi-check text-base"
+                />
+                <i
+                  v-if="option.icon === 'partial'"
+                  class="pi pi-minus"
+                  style="font-size: .8rem"
+                />
+              </div>
+              <div class="flex-1 w-full">
+                {{ option.label }}
+              </div>
+            </div>
+          </PVButton>
+        </div>
+      </div>
+    </PVOverlayPanel>
+  </div>
+</template>
+
+<style scoped lang="scss">
+// Note, these styles are meant to match the size of the PV checkboxes, which we cannot
+// use directly because of the mixed (dash) state here. They appear to be 1.25rem wide.
```

Comment:
Could optionally use variables for this.

Reply:
I'll pass on this for now. However, I'll create a TODO to bring in the RMI PVDesigner codebase in through some kind of linkage/customization scheme, which would make making changes like this easier to handle/upgrade piecemeal. #97